### ISSUE TITLE
SC 1411 -- Contrast of focus state below 3:1

### ISSUE THREAD

**Issue by damiansian** [newjersey/accessibility-pm#10]
## URL 
https://demo.fileyourstatetaxes.org/en/nj/landing-page

## Issue
The contrast of the visible focus indicator is too low in certain use cases. The minimum ratio for the contrast indicator must be 3:1 to the adjacent background. We see lower the 3:1 in certain areas of the UI. 

![Image](https://github.com/user-attachments/assets/d84ca33e-f455-4b88-acb6-81abf19b803a)

## WCAG
[SC 1.4.11: Non-text Contrast (Level AA)](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html#related-focus)

## User impact
Users with low vision benefit from stronger contrast in the user interface. Therefore, a 3:1 contrast minimum is required for the focus indicator of interactive controls. 

### Should be
```css
box-shadow: 0 0 0 5px #21C834, inset 0px 2px 0px rgba (0,0,0,0.15)
```

**Comment by amanifarooque** [comment]
Guidance from CFA design is to use `#005C0F`!

**Comment by damiansian** [comment]
@amanifarooque 

#005C0F creates a very strong contrast on white, and the blue of the disclosure. But luminosity-wise, it's too close to the dark blue of the banner. 

![Contrast testing results showing failure at 1.6:1](https://github.com/user-attachments/assets/75f03f94-3ad5-4bf8-a4db-94cff85feb19)

### Possible solutions
- Create a focus style for the banner with a lighter color, or
- Use #1F883D (recommended)

### Results
#1F883D over #E7F6F8 (light blue of disclosure) = 4.1:1 Pass
#1F883D over #FFFFFF (light blue of disclosure) = 4.5:1 Pass
#1F883D over #112F4E (dark blue of banner) = 3:1 Pass

**Comment by amurthy2** [comment]
@damiansian Hi! I ran some tests on my end and it looks like #1A8CFF might work for focus state (this is close to what USWDS uses and should work across state primary colors and the blue reveals) LMK what you think!

**Comment by damiansian** [comment]
@amurthy2 

That works. Let me know when I can retest :) 

Also, you may consider putting a 2px offset on the focus style like this: 
<img width="165" alt="button with a 2px offset on the focus style's stroke" src="https://github.com/user-attachments/assets/c0f35097-a30b-43de-82c4-edbe98c5f597" />

I find it really makes the focus super clear. Just a suggestion though.

**Comment by amurthy2** [comment]
sounds good! feel free to go ahead and retest!

**Comment by damiansian** [comment]
@amurthy2 

I retested in Stage/Demo and no changes had been made. I cleared cache and tried an incognito window as well. 

When this is ready for me to retest in stage/demo, please mark the ticket status to "Ready to retest". I'm moving to "Needs rework" as I have retested and found the same issue to be present.

**Comment by tristanvanech** [comment]
@damiansian we'll be working on the fix here, not CFA, so @mmazanec22 will give us the go-ahead to retest

**Comment by damiansian** [comment]
@tristanvanech 

Can you please ensure I have the required permissions and credentials to access the other environment?

**Comment by tristanvanech** [comment]
we will post the heroku staging link here once it's ready to test! you won't need any additional access to view and test

**Comment by mmazanec22** [comment]
@amurthy2 here's the PR with the new focus styling: https://github.com/codeforamerica/vita-min/pull/5349

**Comment by tristanvanech** [comment]
@damiansian here's that staging link for ya! https://statefile.pr-5349.getyourrefund-testing.org/en/nj/landing-page

**Comment by damiansian** [comment]
@mmazanec22 

This is looking really solid! I see a bit of overlapping of the focus state and the adjacent text. 

![Blue focus touches adjacent text](https://github.com/user-attachments/assets/8e90a7b5-2273-4111-9387-bb755e07262b)

I'm not sure if you'd be open to the idea that link text can have a different visual focus than controls like buttons, selectboxes, inputs and such. At Adobe we solved for this by adding an additional underline to `a` tags. The CSS here plays nicely with descenders in text (i.e, the squiggly bit of the "g" in the screen shot"). 

![two underlines on link text in keyboard focus state](https://github.com/user-attachments/assets/32e19dc9-00d6-43a7-a974-bab57f252712)

Reference: 
https://react-spectrum.adobe.com/react-spectrum/Link.html

**Comment by mmazanec22** [comment]
@damiansian I'm open to that but it probably requires some time/coordination for the designers on both teams to figure out how they want this to look.

In the meantime, if I reduce the size of the focus highlight, will this be ok to merge?

**Comment by damiansian** [comment]
Passes minimums and looks great! 

<img width="1425" alt="3:1 and higher all around." src="https://github.com/user-attachments/assets/d1ad8fe0-f80d-46b1-8156-ded78c0cb496" />
